search/search_test.go

```diff
@@ -109,14 +110,127 @@ func TestSearchUnit(t *testing.T) {
 			So(err, ShouldEqual, searchErr)
 			So(m, ShouldBeNil)
 		})
+
+		Convey("test Dimension uses default search limit when no limit config provided", func() {
+
+			mockClient := &rchttp.ClienterMock{
+				DoFunc: func(ctx context.Context, req *http.Request) (*http.Response, error) {
+					return &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(searchResp))),
+					}, nil
+				},
+			}
+
+			searchCli := &Client{
+				cli: mockClient,
+				url: "http://localhost:22000",
+			}
+
+			Convey("when dimension search is called", func() {
+				m, err := searchCli.Dimension(ctx, "12345", "time-series", "1", "geography", "Newport", Config{Offset: &offset})
+
+				Convey("then the request is sent with the default limit", func() {
+					So(mockClient.DoCalls(), ShouldHaveLength, 1)
+					q := mockClient.DoCalls()[0].Req.URL.Query()
+					So(q.Get("limit"), ShouldEqual, strconv.Itoa(defaultLimit))
+				})
+
+				Convey("and the expected model is returned", func() {
+					So(err, ShouldBeNil)
+					So(m.Count, ShouldEqual, 1)
+					So(m.Limit, ShouldEqual, 1)
+					So(m.Offset, ShouldEqual, 0)
+					So(m.TotalCount, ShouldEqual, 1)
+					So(m.Items, ShouldHaveLength, 1)
+
+					item := m.Items[0]
+					So(item.Code, ShouldEqual, "6789")
+					So(item.DimensionOptionURL, ShouldEqual, "http://localhost:22000/datasets/12345/editions/time-series/versions/1/dimensions/geography/options/6789")
+					So(item.HasData, ShouldBeTrue)
+					So(item.Label, ShouldEqual, "Newport")
+					So(item.Matches.Label, ShouldHaveLength, 1)
+					So(item.NumberOfChildren, ShouldEqual, 3)
+
+					label := item.Matches.Label[0]
+					So(label.Start, ShouldEqual, 0)
+					So(label.End, ShouldEqual, 6)
+				})
+			})
+		})
+
+		Convey("test Dimension no limit returns error from HTTPClient if it throws an error", func() {
+			mockClient := &rchttp.ClienterMock{
+				DoFunc: func(ctx context.Context, req *http.Request) (*http.Response, error) {
+					return nil, errors.New(clientErrText)
+				},
+			}
+
+			searchCli := &Client{
+				cli: mockClient,
+				url: "http://localhost:22000",
+			}
+
+			Convey("when search is called", func() {
+				m, err := searchCli.Dimension(ctx, "12345", "time-series", "1", "geography", "Newport", Config{Offset: &offset})
+
+				Convey("then the request is sent with the default limit", func() {
+					So(mockClient.DoCalls(), ShouldHaveLength, 1)
+					q := mockClient.DoCalls()[0].Req.URL.Query()
+					So(q.Get("limit"), ShouldEqual, strconv.Itoa(defaultLimit))
```

Comment:
I'd be tempted to put
  `var defaultLimitAsString = strconv.Itoa(defaultLimit)`
at the top, to make this (and the others, below) more readable.

Otherwise, looks good.